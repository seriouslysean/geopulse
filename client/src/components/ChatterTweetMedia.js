import React from 'react';

class ChatterTweetMedia extends React.Component {
  renderPhoto = () => {
    const {
      entity: {
        expanded_url: href,
        media_url_https: src,
        sizes: { large: { h: height, w: width } },
      },
    } = this.props;
    return (
      <a rel="external" href={href} target="_blank">
        <img src={src} height={height} width={width} />
      </a>
    );
  };
  render() {
    const { entity } = this.props;
    let content = '';
    switch (entity.type) {
      case 'photo':
        content = this.renderPhoto();
        break;
    }
    return <div className="chatter__media">{content}</div>;
  }
}

export default ChatterTweetMedia;
