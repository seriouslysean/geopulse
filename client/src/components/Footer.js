import React from 'react';

const Footer = () => (
  <footer className="section section--bg-earth-darkest footer">
    <div className="content-container">
      <p>
        Crafted with love /{' '}
        <a rel="external" href="https://www.codeserious.com" className="footer__link">
          Code Serious
        </a>
      </p>
    </div>
  </footer>
);

export default Footer;
