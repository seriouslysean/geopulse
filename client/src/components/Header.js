import React from 'react';
import { Link, NavLink } from 'react-router-dom';

const Header = () => (
  <header className="section section--bg-sky-dark header">
    <div className="content-container">
      <div className="header__content">
        <Link to="/" className="header__logo">
          <span>
            <img src="./assets/logo-geopulse.svg" alt="Geopulse logo" />
          </span>
          <span>Geopulse</span>
        </Link>
        <nav className="header__navigation">
          <ul>
            <li>
              <NavLink to="/" className="header__link" activeClassName="header__link--active" exact>
                Feed
              </NavLink>
            </li>
            <li>
              <NavLink to="/about" className="header__link" activeClassName="header__link--active">
                About
              </NavLink>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
);

export default Header;
