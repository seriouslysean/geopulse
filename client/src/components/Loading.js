import React from "react";

const Loading = () => (
  <div className="loading">
    <div className="loading__icon" style={{ backgroundImage: "url(./assets/loading.svg)" }} />
  </div>
);

export default Loading;
