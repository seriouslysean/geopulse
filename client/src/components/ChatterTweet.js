import React from "react";
import { getTwitterUserUrl, getTweetMedia, setupTweetText } from "../helpers/chatter";
import ChatterTweetMedia from "./ChatterTweetMedia";

const ChatterTweet = ({ tweet }) => {
  let originalTweet = {};
  if (tweet.retweeted_status) {
    originalTweet = tweet;
    tweet = tweet.retweeted_status;
  }
  const { id, user: { screen_name } } = tweet;
  return (
    <li className="chatter__status">
      {tweet.entities.media &&
        tweet.entities.media.map(entity => {
          return <ChatterTweetMedia key={entity.id} entity={entity} />;
        })}
      <div className="chatter__text" dangerouslySetInnerHTML={{ __html: setupTweetText(tweet) }} />
      <div className="chatter__details">
        {tweet.retweeted_status && (
          <div className="status__rt">
            RT /{" "}
            <a
              rel="external"
              href={getTwitterUserUrl(originalTweet.user.screen_name)}
              title={`Follow ${originalTweet.user.screen_name} on Twitter`}
              target="_blank"
            >
              @{screen_name}
            </a>
          </div>
        )}
        <div className="status__username">
          <a
            rel="external"
            href={getTwitterUserUrl(screen_name)}
            title={`Follow ${screen_name} on Twitter`}
            target="_blank"
          >
            @{screen_name}
          </a>
        </div>
      </div>
    </li>
  );
};

export default ChatterTweet;
