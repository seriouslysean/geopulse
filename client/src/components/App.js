import React from "react";
import { connect } from "react-redux";
import { renderRoutes } from "react-router-config";
import { Helmet } from "react-helmet";
import Header from "./Header";
import Footer from "./Footer";

export const App = ({ route }) => {
  return (
    <div className="app">
      <Helmet>
        <title>Geopulse</title>
        <meta property="og:title" content="Geopulse" />
      </Helmet>

      <Header />

      {renderRoutes(route.routes)}

      <Footer />
    </div>
  );
};

export default {
  component: App
};
