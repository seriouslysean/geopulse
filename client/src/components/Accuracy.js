import React from "react";
import { connect } from "react-redux";
import { getGeolocation, setGeolocation } from "../actions/geolocation";
import { getWeather, setWeather } from "../actions/weather";
import { getPhotos, setPhotos } from "../actions/photos";
import { getChatter, setChatter } from "../actions/chatter";
import { getVideos, setVideos } from "../actions/videos";
import Loading from "../components/Loading";

class Accuracy extends React.Component {
  handleImproveAccuracy = e => {
    this.props
      .resetGeolocation()
      .then(() => {
        this.props.getGeolocation();
      })
      .catch(() => {
        console.error("Unable to improve accuracy");
      });
  };
  renderContent = () => {
    const { longitude, latitude } = this.props.geolocation;
    return (
      <div className="content">
        <p className="accuracy__intro">
          Your location has been approximated using your IP address. If you'd like to improve the accuracy, please click
          the button below.
        </p>

        <button onClick={this.handleImproveAccuracy} className="button accuracy__button">
          Improve Accuracy
        </button>

        <div className="accuracy_geolocation">
          {latitude} / {longitude}
        </div>
      </div>
    );
  };
  render() {
    return (
      <section className="section section--spacing section--bg-sky-light accuracy">
        <div className="content-container">
          <div className="accuracy__content">
            <h1 className="section__title">Accuracy</h1>

            {this.props.ready && this.props.geolocation ? this.renderContent() : <Loading />}
          </div>
        </div>
      </section>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  resetGeolocation: () => {
    return Promise.all([
      dispatch(setGeolocation()),
      dispatch(setWeather()),
      dispatch(setPhotos()),
      dispatch(setChatter()),
      dispatch(setVideos())
    ]);
  },
  getGeolocation: () => dispatch(getGeolocation())
});

export default connect(undefined, mapDispatchToProps)(Accuracy);
