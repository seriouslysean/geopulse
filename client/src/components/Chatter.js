import React from "react";
import Loading from "./Loading";
import ChatterTweet from "./ChatterTweet";

class Chatter extends React.Component {
  renderChatter = () => {
    return (
      <ul className="chatter__list">
        {this.props.chatter.collection.map(status => {
          return <ChatterTweet key={status.id} tweet={status} />;
        })}
      </ul>
    );
  };
  renderContent = () => {
    return (
      <div className="content">
        {this.props.chatter.collection.length ? this.renderChatter() : <p className="text--center">No chatter found</p>}
      </div>
    );
  };
  render() {
    return (
      <section className="section section--spacing section--bg-earth-dark chatter">
        <div className="content-container content-container--grid-spacing">
          <div className="chatter__content">
            <h1 className="section__title">Chatter</h1>

            {this.props.ready && this.props.chatter ? this.renderContent() : <Loading />}

            <a rel="external" href="https://www.twitter.com/" className="section__attribution" target="_blank">
              Powered by Twitter
            </a>
          </div>
        </div>
      </section>
    );
  }
}

export default Chatter;
