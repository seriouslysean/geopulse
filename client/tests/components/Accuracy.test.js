import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import geolocationProp from "../../../data/fixtures/geolocation";
import { Accuracy } from "../../src/components/Accuracy";

test("should show Accuracy loading correctly", () => {
  const wrapper = shallow(<Accuracy />);
  expect(wrapper).toMatchSnapshot();
});

test("should render Accuracy correctly", () => {
  const wrapper = shallow(<Accuracy ready={true} geolocation={geolocationProp} />);
  expect(wrapper).toMatchSnapshot();
});
