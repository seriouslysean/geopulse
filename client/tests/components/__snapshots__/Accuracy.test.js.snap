// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Accuracy correctly 1`] = `
<section
  className="section section--spacing section--bg-sky-light accuracy"
>
  <div
    className="content-container"
  >
    <div
      className="accuracy__content"
    >
      <h1
        className="section__title"
      >
        Accuracy
      </h1>
      <div
        className="content"
      >
        <p
          className="accuracy__intro"
        >
          Your location has been approximated using your IP address. If you'd like to improve the accuracy, please click the button below.
        </p>
        <button
          className="button accuracy__button"
          onClick={[Function]}
        >
          Improve Accuracy
        </button>
        <div
          className="accuracy_geolocation"
        >
          39.9339456
           / 
          -75.1548441
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`should show Accuracy loading correctly 1`] = `
<section
  className="section section--spacing section--bg-sky-light accuracy"
>
  <div
    className="content-container"
  >
    <div
      className="accuracy__content"
    >
      <h1
        className="section__title"
      >
        Accuracy
      </h1>
      <Loading />
    </div>
  </div>
</section>
`;
